src/cis0.rs

```diff
@@ -0,0 +1,251 @@
+        let mut results = Vec::new();
+        for _ in 0..len {
+            results.push(SupportResult::deserial(source)?)
+        }
+        Ok(Self { results })
+    }
+}
+
+impl contracts_common::Serial for SupportsQueryResponse {
+    fn serial<W: contracts_common::Write>(&self, out: &mut W) -> Result<(), W::Err> {
+        (self.results.len() as u16).serial(out)?;
+        for result in &self.results {
+            result.serial(out)?;
+        }
+        Ok(())
+    }
+}
+
+/// Identifier for a smart contract standard.
+#[non_exhaustive]
+#[derive(Clone, Debug)]
+pub enum StandardIdentifier {
+    CIS0,
+    CIS1,
+    CIS2,
+    Other(String),
+}
+
+impl std::fmt::Display for StandardIdentifier {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            StandardIdentifier::CIS0 => f.write_str("CIS-0"),
+            StandardIdentifier::CIS1 => f.write_str("CIS-1"),
+            StandardIdentifier::CIS2 => f.write_str("CIS-2"),
+            StandardIdentifier::Other(s) => f.write_str(s),
+        }
+    }
+}
+
+#[derive(Error, Debug, Clone)]
+/// The standard name is invalid and could not be parsed.
+#[error("Invalid CIS standard name.")]
+pub struct InvalidStandardName;
+
+impl std::str::FromStr for StandardIdentifier {
+    type Err = InvalidStandardName;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "CIS-0" => Ok(Self::CIS0),
+            "CIS-1" => Ok(Self::CIS1),
+            "CIS-2" => Ok(Self::CIS2),
+            other if other.is_ascii() && other.len() <= 255 => Ok(Self::Other(other.into())),
+            _ => Err(InvalidStandardName),
+        }
+    }
+}
+
+impl contracts_common::Serial for StandardIdentifier {
+    fn serial<W: contracts_common::Write>(&self, out: &mut W) -> Result<(), W::Err> {
+        match self {
+            StandardIdentifier::CIS0 => {
+                out.write_u8(5)?; // length
+                out.write_all(b"CIS-0")
+            }
+            StandardIdentifier::CIS1 => {
+                out.write_u8(5)?; // length
+                out.write_all(b"CIS-1")
+            }
+            StandardIdentifier::CIS2 => {
+                out.write_u8(5)?; // length
+                out.write_all(b"CIS-2")
+            }
+            StandardIdentifier::Other(s) => {
+                out.write_u8(s.len() as u8)?;
+                out.write_all(s.as_bytes())
+            }
+        }
+    }
+}
+
+#[derive(Error, Debug)]
+/// Errors that may occur when querying a contract.
+pub enum SupportsError {
+    #[error("The name of the contract is not valid and thus the contract does not support CIS0.")]
+    ContractNameInvalid,
+    #[error("Parameter size exceeds maximum allowed. Too many ids.")]
+    InvalidParameter,
+    #[error("Query error: {0}")]
+    QueryError(#[from] super::v2::QueryError),
+    #[error("Contract reject.")]
+    ContractReject,
+    #[error("No return. This is a V0 contract.")]
```

Comment:
Maybe add that: Thus the contracts does not support CIS0

Reply:
Makes sense.